strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactControllerTest.java

```diff
@@ -125,4 +139,77 @@ public void testPackageCommonFlow(@NpmRepository(repositoryId = REPOSITORY_RELEA
+                             "@lifaon/observables:1.6.0" })
+    public void packageNameAcceptanceTest(String packageNameWithVersion,
+                                          @NpmRepository(repositoryId = REPOSITORY_RELEASES)
+                                          Repository repository,
+                                          TestInfo testInfo)
+            throws Exception
+    {
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        //Splitting each string from paramterized test array to packageId and packageVersion
+        final String[] packageDetails = packageNameWithVersion.split(":");
+        final String packageId = packageDetails[0];
+        final String packageVersion = packageDetails[1];
+
+        //Manually generating each artifacts
+        generatorBasePath = Paths.get(propertiesBooter.getVaultDirectory(),
+                                      ".temp",
+                                      testInfo.getTestClass().get().getSimpleName(),
+                                      testInfo.getTestMethod().get().getName()).toAbsolutePath().normalize();
+
+        NpmArtifactGenerator artifactGenerator = new NpmArtifactGenerator(generatorBasePath);
+        Path artifact = artifactGenerator.generateArtifact(packageId, packageVersion);
+
+        NpmArtifactCoordinates coordinates = NpmArtifactCoordinates.of(packageId, packageVersion);
+
+        Path publishJsonPath = artifact.resolveSibling("publish.json");
+        byte[] publishJsonContent = Files.readAllBytes(publishJsonPath);
+
+        //Publish
+        String url = getContextBaseUrl() + "/storages/{storageId}/{repositoryId}/{artifactId}";
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+                .body(publishJsonContent)
+                .when()
+                .put(url, storageId, repositoryId, coordinates.getId())
+                .peek()
+                .then()
+                .statusCode(HttpStatus.OK.value());
+
+        //View
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .get(url, storageId, repositoryId, coordinates.getId())
+                .peek()
+                .then()
+                .assertThat()
+                .body("name", equalTo(packageId))
+                .and()
+                .assertThat()
+                .body("versions." + "'" + packageVersion + "'" +".version", equalTo(packageVersion))
+                .statusCode(HttpStatus.OK.value());
+
+        //Download
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .get(url, storageId, repositoryId, coordinates.toResource())
+                .prettyPeek()
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .assertThat()
+                .header(HttpHeaders.CONTENT_LENGTH, equalTo(String.valueOf(Files.size(artifact))));
+    }
+
+}
```

Comment:
Cosmetic: Please, make sure that all source/resource files end with a blank line! Thanks! :)